### DISCUSSION TITLE
UGCS connects to the right COM port but cannot understand TinyG

### DISCUSSION THREAD

**Discussion by Bluemetal09** [winder/Universal-G-Code-Sender#1895] [Help] [answered]
After installing UGCS in a Windows machine that is connected to a TinyG (fv0.97: fb440.20) Shapeoko (and runs with other g-code senders) and trying to connect to the controller it tells me that:
**** Connected to COM2 @ 115200 **** 
while the big gray button on the top of the Controller State (DRO) window keeps saying "Connecting". 
Then it displays the message you can see below. Basically, the response one sees when connecting to TinyG with other g-code sender packages but here it never fully connects.
If I try to use the setup wizard, after choosing TinyG, COM2 and 115200, it tells me that it "Connected to TinyG unknown version and that the setup wizard is not supported for this controller".

Is this the case, i.e. , that UCGS is incompatible with TinyG 0.97 (440.20) or I am making some rooky mistake somewhere? Thank you in advance for any guidance.

![Error message UGS](https://user-images.githubusercontent.com/1641266/171309816-ba103f22-0ffc-414b-8484-8c0b8b9da4eb.PNG)

**Comment by breiler** [comment]
It worked fine on my computer and I can not reproduce the error using the same firmware version:
<img width="837" alt="image" src="https://user-images.githubusercontent.com/8962024/171350730-0251ae17-b519-4981-b454-1f1447e762de.png">

The message `STATUS READY` is missing in the response from the controller in your screen shot. Try to reboot the controller by unpowering/power it and let it start up for three seconds. Then connect with UGS.

Also note that the edge version of TinyG (449.01) doesn't work in UGS. They have changed the protocol and also forgot to add the controller state in the status report. UGS have no way of knowing in which state the controller is in.

**Comment by Bluemetal09** [comment, marked as answer]
Thank you for the prompt response. I believe the issue resides on a faulty board. The reset and reboot did not help. I also tried with a mac and behaved exactly the same - pointing to a board issue. Only after a factory default reset ($defa=1) did the board connect properly with UGCS and got the exact same response you've got including the SYSTEM READY message.
Changing any of the board parameters would prevent new connections though. Only another factory default reset was performed the connection would work once again. The board was also not storing certain parameters while keeping others. It needs to be replaced.
